src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1271,9 +1271,14 @@ public function getHost()
 
         // as the host can come from the user (HTTP_HOST and depending on the configuration, SERVER_NAME too can come from the user)
         // check that it does not contain forbidden characters (see RFC 952 and RFC 2181)
+        // and if it honors size limits (see RFC 1035 - 2.3.4) taking care about IDNA (see RFC 5890)
         // use preg_replace() instead of preg_match() to prevent DoS attacks with long host names
-        if ($host && '' !== preg_replace('/(?:^\[)?[a-zA-Z0-9-:\]_]+\.?/', '', $host)) {
-            throw new \UnexpectedValueException(sprintf('Invalid Host "%s"', $host));
+        if ($host) {
+            if ('' !== preg_replace('/(?:^\[)?[a-zA-Z0-9-:\]_]+\.?/', '', $host) ||
+                0 !== preg_match('/(?:^\[)?[a-zA-Z0-9-:\.\]_]{256,}|[a-zA-Z0-9-:\]_]{64,}\.?/', idn_to_ascii($host))
```

Comment:
You're right @dunglas, thank you. I'm updating the tests and this logic in order to check if ```idn_to_ascii()``` returns false (while invalid IDNs). Later I'll check ```filter_var()``` with ```FILTER_VALIDATE_URL``` based in your PR.